scriptworker/artifacts.py

```diff
@@ -135,6 +132,11 @@ def guess_content_type_and_encoding(path):
         str: the content type of the file
 
     """
+    #  TODO: is mimetypes needed? Maybe we can depend purely on our own _EXTENSION_TO_MIME_TYPE mapping
```

Comment:
I thought about this a bit too. We mostly upload a known set of file extensions, and falling back to `application/binary` isn't the worst. But there are no guarantees we won't upload a new file extension, or a dozen, in the future. So we could:

- have a small subset of known file extensions and potentially adjust this map every time we add a new extension
- have a full set of known file extensions, and potentially adjust this map more rarely, if we upload a brand new file extension that doesn't exist in the full set
- have a small subset of file extensions to override the system mimetypes, but fall back to the system definitions. This is what we have here.

In each of these cases, we could require a scriptworker release, or we could configure the known file extensions via the `scriptworker.yaml` config file per scriptworker instance.

In practice I'm not sure the win is large enough to warrant further change, but I could be wrong.

Reply:
Part of why this is so challenging is that we pass `content_type` to Taskcluster, at which point any number of things consume it. Unfortunately, this means that all those downstream components depend on the content-type detection of `scriptworker`.

In the mean time, I think that option #3 (have small subset, fall back to system definitions) makes the most sense. Though it's ~inconsistent, it seems to have generally worked so far.

-----

Hmm, we might have a 4th option. If the only reason we're overriding mimetypes is for the gzip logic (and **not** for downstream consumption, e.g.: browsers when clicking artifacts), then we can just do:
* Use mimetypes + extension-matching to determine if something is gzippable. But, when providing `content_type` to TC, just give it the system mimetype without overriding.

I don't think that this discussion is blocking the PR, so I'll merge it :)

Reply:
> Use mimetypes + extension-matching to determine if something is gzippable. But, when providing content_type to TC, just give it the system mimetype without overriding.

I think that means you end up with files like blah.log or foo.asc that are cleartext, but you have to download to read instead of being able to read them in your browser.

Reply:
Ew :)
Yeah, I agree that option 3 is probably our best bet :smile: